src/main/java/com/rocketden/main/controller/v1/ProblemController.java

```diff
@@ -0,0 +1,36 @@
+package com.rocketden.main.controller.v1;
+
+import com.rocketden.main.dto.problem.Problem;
+import com.rocketden.main.dao.problem.ProblemRepository;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class ProblemController extends BaseRestController {
+
+	@Autowired
+	private ProblemRepository problemRepository;
+  
+  @PostMapping("/addProblem")
+  public @ResponseBody Problem addNewProblem (@RequestParam String name, 
+    @RequestParam String description) {
```

Comment:
This might be a nit, but is it possible to change these from RequestParam to RequestBody annotations? This gets a bit deep into REST API design where you have things like query variables, path variables, and body variables (these two sites [[1]](https://stackoverflow.com/questions/14551194/how-are-parameters-sent-in-an-http-post-request) [[2]](https://softwareengineering.stackexchange.com/questions/332637/rest-webservices-post-put-which-type-of-parameter-should-i-choose-and-whe) give an overview of the differences between them and [this](https://stackoverflow.com/questions/28039709/what-is-difference-between-requestbody-and-requestparam) details how to use them with Spring), but the gist from what I've read is that GET tends to use query params whereas POST tends to use body params. However, this change would affect a lot of the tests so we can always revisit this later (esp since you already spent a ton of time on this PR lol).

Reply:
This is a good point. It was a lot of changes, but I thought it was important so I replaced the two `@RequestParam` parameters with a single `@RequestBody` variable. I had to add a function in `util/Utility.java` that converted the `x-www-form-urlencoded` body format to a `Map<String, String>` mapping the keys to values, but I managed to do so. I also had to reformat the testing as you mentioned, and putting the body in a String turned out to be the easiest way to do this, but I got this to work successfully as well.

Reply:
Note: There may be other `replace(...)` things that we need to include later; right now I only have the following two:
```
pair[1].replace("+", " ");
pair[1].replace("%0A", "\n");
```
However, I think we can add them as they come up. I thought there would be a more elegant solution to convert the body into a proper format, but this is the only attempt that worked.

Reply:
I feel bad cause I think one of the links I shared did mention `x-www-form-urlencoded` which might not be best suited for our use case; there should be an easier way where you accept JSON by just putting `@ResponseBody Problem` as the parameter instead of String, similar to the POST example in this tutorial: https://spring.io/guides/tutorials/bookmarks/

That's definitely my bad and we can deal with the details of it later, don't want to keep burdening you on this PR haha

Reply:
Ah, I tried doing `@ResponseBody Problem`, but it didn't work -- possibly because I didn't have `application/json`. I'll try this on another branch.

Reply:
It might have to do with the way you send in the POST request using Postman, I think under Body you have to select raw -> JSON in order to match the format on the client end.